test/extensions/access_loggers/grpc/tls_grpc_access_log_integration_test.cc

```diff
@@ -0,0 +1,508 @@
+#include <memory>
+
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/config/core/v3/address.pb.h"
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/extensions/access_loggers/file/v3/file.pb.h"
+#include "envoy/extensions/access_loggers/grpc/v3/als.pb.h"
+#include "envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.pb.h"
+#include "envoy/service/accesslog/v3/als.pb.h"
+
+#include "source/common/buffer/zero_copy_input_stream_impl.h"
+#include "source/common/config/api_version.h"
+#include "source/common/grpc/codec.h"
+#include "source/common/grpc/common.h"
+#include "source/common/network/utility.h"
+#include "source/common/version/version.h"
+#include "source/extensions/filters/listener/tls_inspector/tls_inspector.h"
+#include "source/extensions/transport_sockets/tls/context_manager_impl.h"
+#include "source/extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/integration.h"
+#include "test/integration/ssl_utility.h"
+#include "test/integration/utility.h"
+#include "test/mocks/secret/mocks.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+void clearPort(envoy::config::core::v3::Address& address) {
```

Comment:
This includes a lot of code copied from `tcp_grpc_access_log_integration_test.cc`, this should either be:
- extract to a base header file with common part (like `grpc_access_log_integration_test_base.h`), or
- just add your new test class and tests into `tcp_grpc_access_log_integration_test.cc`.

Reply:
Thanks. Yes, that is true. I will take another look at integrating with `tcp_grpc_access_log_integration_test.cc` next week when I get back after the 22nd.

Reply:
Hi,
I just made new a commit. CI Pipeline ran successfully.

In new commit:

- new tls test cases are added to (integrated with) `tcp_grpc_access_log_integration_test.cc`
- `tls_grpc_access_log_integration_test.cc` is removed.